src/test/scala/mesosphere/marathon/integration/UpgradeIntegrationTest.scala

```diff
@@ -0,0 +1,231 @@
+package mesosphere.marathon
+package integration
+
+import java.io.File
+import java.net.URL
+import java.nio.file.Files
+
+import akka.actor.{ ActorSystem, Scheduler }
+import akka.stream.Materializer
+import mesosphere.{ AkkaIntegrationTest, WhenEnvSet }
+import mesosphere.marathon.integration.setup._
+import mesosphere.marathon.io.IO
+import mesosphere.marathon.state.PathId
+import org.apache.commons.io.FileUtils
+import org.scalatest.concurrent.Eventually
+
+import scala.concurrent.ExecutionContext
+import scala.sys.process.Process
+
+/**
+  * This integration test starts older Marathon versions one after another and finishes this upgrade procedure with the
+  * current build. In each step we verfiy that all apps are still up and running.
+  */
+@IntegrationTest
+class UpgradeIntegrationTest extends AkkaIntegrationTest with MesosClusterTest with ZookeeperServerTest with Eventually {
+
+  // Integration tests using docker image provisioning with the Mesos containerizer need to be
+  // run as root in a Linux environment. They have to be explicitly enabled through an env variable.
+  val envVar = "RUN_MESOS_INTEGRATION_TESTS"
```

Comment:
I suggest moving this envVar in one of the basic IT test objects. all tests should use the same ;)